scio-core/src/main/scala/com/spotify/scio/ScioResult.scala

```diff
@@ -54,7 +60,49 @@ class ScioResult private[scio] (val internal: PipelineResult,
   def accumulatorValuesAtSteps[T](acc: Accumulator[T]): Map[String, T] =
     getAggregatorValues(acc).flatMap(_.getValuesAtSteps.asScala).toMap
 
+  /** Save metrics of the finished pipeline to a file. */
+  def saveMetrics(filename: String): Unit = {
+    require(isCompleted, "Pipeline has to be finished to save metrics.")
+
+    val mapper = new ObjectMapper()
+    mapper.registerModule(DefaultScalaModule)
+
+    val out = IOChannelUtils.create(filename, MimeTypes.TEXT)
+
+    try {
+      val totalValues = accumulators
+        .map(acc => AccumulatorValue(acc.name, accumulatorTotalValue(acc)))
+
+      val stepsValues = accumulators
+        .map(acc => AccumulatorStepsValue(acc.name,
+          accumulatorValuesAtSteps(acc).map(a => AccumulatorStepValue(a._1, a._2))))
+
+      val options = this.pipeline.getOptions
+      val metrics = Metrics(scioVersion,
+                            scalaVersion,
+                            options.as(classOf[ApplicationNameOptions]).getAppName,
+                            options.as(classOf[DataflowPipelineOptions]).getJobName,
```

Comment:
Not sure if this is completely safe. It works fine for local mode (and dataflow), not sure about other runners.